[Bug]: Provider : error: the server doesn't have a resource type "storagecluster"

### Description
Test case test_change_reclaim_policy_of_pv failing with the error 

```
E               ocs_ci.ocs.exceptions.CommandFailed: Error during execution of command: oc patch storagecluster ocs-storagecluster -n openshift-storage-client --type json --patch  '[{ "op": "replace", "path": "/spec/enableCephTools", "value": true }]'.
E               Error is error: the server doesn't have a resource type "storagecluster"



ocs_ci/utility/utils.py:722: CommandFailed
```

in provider-client configuration.


### Steps to Reproduce
1. Run the test test_change_reclaim_policy_of_pv on provider - client configuration


### Actual Behavior
the server doesn't have a resource type "storagecluster"

### Expected Behavior
Provider clsuters should be used automatically for storagecluster kind.
### Impact (likelihood of reproduction, impact on the cluster and on other tests, etc.)
Test case fails
### Screenshots (if applicable)

### Environment
- Test Suite(s): https://reportportal-ocs4.apps.dno.ocp-hub.prod.psi.redhat.com/ui/#ocs/launches/795/30881/1517012/1517273/log
- Platform(s): IBM Cloud Bare Metal
- Version(s): ODF and OCp 4.18

### Additional Context

--- a/ocs_ci/ocs/ocp.py
+++ b/ocs_ci/ocs/ocp.py
@@ -90,7 +90,7 @@ class OCP(object):
         if (
             (not cluster_kubeconfig)
             and config.multicluster
-            and config.ENV_DATA.get("odf_provider_mode_deployment", False)
+            and "hci_" in config.ENV_DATA["platform"]
             and len(config.get_provider_cluster_indexes()) == 1
             and kind.lower() in constants.PROVIDER_CLUSTER_RESOURCE_KINDS
         ):

